src/routes/safe/components/AllTransactions/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { currentPageSelector, newTransactionsCurrentPageSelector } from '../../store/selectors/newTransactions'
+import { useFetchNewTransactions } from '../../container/hooks/useFetchNewTransactions'
+import { ButtonLink, Loader } from '@gnosis.pm/safe-react-components'
+import { nextPage, previousPage } from '../../store/actions/transactionsNew/pagination'
+import { Transaction } from '../../store/models/types/transactions'
+import { safeParamAddressFromStateSelector } from '../../store/selectors'
+
+const Transactions = (): React.ReactElement => {
+  const dispatch = useDispatch()
+  const transactions = useSelector(newTransactionsCurrentPageSelector)
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const { currentPage, maxPages, limit, offset } = useSelector(currentPageSelector)
+  useFetchNewTransactions({ safeAddress, offset, limit })
+
+  // TODO: Remove this once we implement infinite scroll
+  const nextPageButtonHandler = () => {
+    dispatch(nextPage())
+  }
+
+  const previousPageButtonHandler = () => {
+    dispatch(previousPage())
+  }
+
+  if (!transactions) return <div>No txs available for safe</div>
+
+  if (!transactions.length) return <Loader size="lg" />
+
+  return (
+    <>
+      {transactions.map((tx: Transaction) => {
+        let txHash = ''
```

Comment:
can we normalize this after fetching?

Reply:
still not addressed

Reply:
could you clarify what's needed here?

Reply:
can you normalize the data after fetching so we don't have to do this if else thing?

Reply:
so you get these from the backend:
```
[{ hash: '0x' }, { txHash: '0x' }]
```
and then you can do something like
```
const buildTransaction = (tx) =>({ txHash: tx.hash || tx.txHash })

transactions.map(buildTransaction)
```

this way in the redux store we have hash stored in the same key and we can get rid of this if statement